src/anonymize_proxy.ts

```diff
@@ -1,18 +1,18 @@
-const { Server } = require('./server');
-const { nodeify } = require('./utils/nodeify');
+import net from 'net';
+import http from 'http';
+import { Buffer } from 'buffer';
+import { URL } from 'url';
+import { Server } from './server';
+import { nodeify } from './utils/nodeify';
 
 // Dictionary, key is value returned from anonymizeProxy(), value is Server instance.
-const anonymizedProxyUrlToServer = {};
+const anonymizedProxyUrlToServer: Record<string, Server> = {};
 
 /**
  * Parses and validates a HTTP proxy URL. If the proxy requires authentication, then the function
  * starts an open local proxy server that forwards to the upstream proxy.
- * @param proxyUrl
- * @param callback Optional callback that receives the anonymous proxy URL
- * @return If no callback was supplied, returns a promise that resolves to a String with
- * anonymous proxy URL or the original URL if it was already anonymous.
  */
-const anonymizeProxy = (proxyUrl, callback) => {
+export const anonymizeProxy = (proxyUrl: string, callback?: (error: Error | null) => void): Promise<string> => {
```

Comment:
What comment are you referring to? This always returns a Promise.

> maybe we could move the nodeify call to the callsite instead of the method definition?

wdym?

Reply:
This one: 

> @return If no callback was supplied, returns a promise that resolves to a String with
> * anonymous proxy URL or the original URL if it was already anonymous.

Which should be caused by the `nodeify` call that turns the promise function to a function with callback, and if the callback is provided, it won't return the value, but feed it into the callback argument instead.

Maybe I got it wrong, but if that is the case, I though we could move the `nodeify` call from inside this method to the call site, so to where the `anonymizeProxy` is used (and where the callback style signature might be required).

Reply:
> Which should be caused by the nodeify call that turns the promise function to a function with callback, and if the callback is provided, it won't return the value, but feed it into the callback argument instead.

Well, in the previous code it was always returning a Promise as well 🤔 Should we change this?

> Maybe I got it wrong, but if that is the case, I though we could move the nodeify call from inside this method to the call site, so to where the anonymizeProxy is used (and where the callback style signature might be required).

Well the issue is that it's used outside this package as well.

Reply:
> Well the issue is that it's used outside this package as well.

I see, ok then let's keep it as it is.